Major refactoring

IvarsKarpics opened the pull request:
This PR concludes major refactoring with following changes:

* removed HardwareRepository submodule
* all file names are now snake_case
* brick names are also in snake_case and corresponding class is in CamelCase. For example tree_brick.py contains class TreeBrick
* renamed few bricks to better names. For example HutchMenuBrick -> sample_control_brick.py, CameraBrick.py -> sample_view_brick.py, OnlineProcessingPreviewBrick -> online_processing_results_brick.py
* moved example gui config file from examples to configuration
* removed app directory and moved mxcube launch script to scripts directory.


TODO: 

* Update readme
* Fix pytest and travis

jordiandreu commented:
Hi @IvarsKarpics , thanks for the large refactoring. I see no blocking issues to merge these changes into master.

Just one comment about the repository structure is that all relevant code should be in mxcube/mxcubeqt (for instance, the scripts folder) and let all repository configuration files at the mxcube level. I would propose this change on the following PR regarding the setup.py configuration so, this PR is fine for me.

IvarsKarpics commented:
Hi @jordiandreu 

I would suggest to keep mxcubeqt directory clean and use just for the mxcubeqt package. Packaging with python also suggest it: https://packaging.python.org/tutorials/packaging-projects/

jordiandreu commented:
Hi Ivars,
I agree, but let me distinguish here the two types of files we currently have in the scripts folder: helper bash files or test files should be of course not in the package folder but the application script file (mxcube_app.py) used to start de mxcube application have to be there. In this way it is possible to define the application as an entry point in the setup.py.
Also in the case of the configuration files, for instance the mockup gui configuration file, if it is included in the package foilder then can be accessed from the mxcube code by using the package_resources module, which makes its access independent of the installation folder.

IvarsKarpics commented:
Hi @jordiandreu 

Sorry, I did not think about that. My comments:
* why not use __init__.py and __main__.py to define entry point. ref: https://setuptools.readthedocs.io/en/latest/userguide/entry_point.html
* current app/mxcube.py is not pretty. I would suggest to drop all the paths to configuration and if someone starts mxcube from console (without env variables) a blank mxcube with clean gui should be launched.
Personally I would use a script and do something like: 
```
from mxcubeqt import create_app
app = create_app()
```

jordiandreu commented:
Hi Ivars,

> why not use init.py and main.py to define entry point. 

Yes, we can use init.py or main.py. This it has no implication regarding the entry_point definition (the entry point maps to any method defined in the python package). However it is true that the selection of the main.py or init.py will have an impact on how to start the application directly from console.

> current app/mxcube.py is not pretty. I would suggest to drop all the paths to configuration and if someone starts mxcube from console (without env variables) a blank mxcube with clean gui should be launched.

Indeed. I would also agree on dropping as much as possible all configuration via environment variables and move to entry_point with several options. For this, I would also suggest to replace the current parsing libraries (optparse) by [click](https://click.palletsprojects.com/en/7.x/).

In addition, we could also provide a mockup gui by default which could be launched like:
`<entry_point> --mockup`

IvarsKarpics commented:
Hi @jordiandreu 

Thanks for productive suggestions. Last changes:
* added __main__.py and changed entry point in setup py.
* removed hard coded configuration paths from __main__.py
* added --mockupMode command line argument to run mockup config.
* moved mockup config to mxcubeqt directory. It is used if --mockupMode argument is used.

rhfogh commented on docs/source/how_to_create_qt_brick.rst:
```diff
@@ -8,9 +8,9 @@ How to create GUI brick in Qt version
 .. code-block:: python
 
-   import QtImport
+   import qt_import
```
change import statements also in this file?

rhfogh commented on docs/source/how_to_create_qt_brick.md:
````diff
@@ -6,9 +6,9 @@
 
 ```
-   import QtImport
+   import qt_import
````
Change import statements also in this file?

rhfogh commented on mxcubeqt/utils/qt_import.py:
```diff
@@ -36,22 +36,22 @@
 could be all replaced by::
 
-   from QtImport import *
+   from qt_import import *
```
Do theese comments need updating?

IvarsKarpics replied:
I renamed QtImport to qt_import. I will update all readme files after accepting this PR

rhfogh replied:
Cool.
